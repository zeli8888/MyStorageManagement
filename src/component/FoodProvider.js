import React, { Component, useState, useEffect } from 'react';
import IngredientService from '../service/IngredientService'
import DishService from '../service/DishService'

export const FoodContext = React.createContext();  //exporting context object

const FoodProvider = ({ children }) => {
    const [ingredients, setIngredients] = useState([]);
    const [dishes, setDishes] = useState([]);

    useEffect(() => {
        IngredientService.getAllIngredients().then(response => { setIngredients(response.data) })
        DishService.getAllDishes().then(response => { setDishes(response.data) })
    }, []);

    return (
        <FoodContext.Provider value={
            {
                ingredients: ingredients,
                setIngredients: setIngredients,
                dishes: dishes,
                setDishes: setDishes
            }}>
            {children
                //this indicates that all the child tags with MyProvider as Parent can access the global store  
            }
        </FoodContext.Provider>)
}

export default FoodProvider;