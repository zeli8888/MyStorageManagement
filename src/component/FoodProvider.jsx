import React, { useState, useEffect } from 'react';
import ingredientService from '../service/IngredientService'
import dishService from '../service/DishService'

export const FoodContext = React.createContext();  //exporting context object

const FoodProvider = ({ children }) => {
    const [allIngredients, setAllIngredients] = useState([]);
    const [allDishes, setAllDishes] = useState([]);

    useEffect(() => {
        ingredientService.getAllIngredients().then(response => { setAllIngredients(response.data) }).catch(error => { console.log(error) });
        dishService.getAllDishes().then(response => { setAllDishes(response.data) }).catch(error => { console.log(error) });
    }, []);

    return (
        <FoodContext.Provider value={
            {
                allIngredients: allIngredients,
                setAllIngredients: setAllIngredients,
                allDishes: allDishes,
                setAllDishes: setAllDishes
            }}>
            {children
                //this indicates that all the child tags with MyProvider as Parent can access the global store  
            }
        </FoodContext.Provider>)
}

export default FoodProvider;