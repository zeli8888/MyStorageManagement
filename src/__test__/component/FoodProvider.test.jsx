import { describe, expect, test, vi, beforeEach } from 'vitest';
import { render, screen, cleanup } from '@testing-library/react';
import FoodProvider, { FoodContext } from '../../component/FoodProvider';
import ingredientService from '../../service/IngredientService'
import dishService from '../../service/DishService'
import { act } from 'react';
import '@testing-library/jest-dom/vitest'

// Mock external services
vi.mock('../../service/IngredientService', () => ({
    default: {
        getAllIngredients: vi.fn(),
    },
}));

vi.mock('../../service/DishService', () => ({
    default: {
        getAllDishes: vi.fn(),
    },
}));

// Mock data for API responses
const mockIngredients = [{ id: 1, name: 'Tomato' }];
const mockDishes = [{ id: 1, name: 'Salad' }];

describe('FoodProvider', () => {
    beforeEach(() => {
        // Reset all mocks before each test
        vi.clearAllMocks();

        // Setup mock implementations
        ingredientService.getAllIngredients.mockResolvedValue({ data: mockIngredients });
        dishService.getAllDishes.mockResolvedValue({ data: mockDishes });
    });
    afterEach(cleanup)

    test('matches snapshot', async () => {
        // Arrange
        let container;
        await act(async () => {
            ({ container } = render(
                <FoodProvider>
                    <div>Test Child</div>
                </FoodProvider>
            ));
        });

        // Assert
        expect(container).toMatchSnapshot();
    });

    test('provides context values to children', async () => {
        // Arrange
        const TestComponent = () => {
            return (
                <FoodContext.Consumer>
                    {(value) => (
                        <div>
                            <span data-testid="ingredients-count">{value.allIngredients.length}</span>
                            <span data-testid="dishes-count">{value.allDishes.length}</span>
                        </div>
                    )}
                </FoodContext.Consumer>
            );
        };

        // Act
        await act(async () => {
            render(
                <FoodProvider>
                    <TestComponent />
                </FoodProvider>
            );
        });

        // Assert
        expect(await screen.findByTestId('ingredients-count')).toHaveTextContent('1');
        expect(await screen.findByTestId('dishes-count')).toHaveTextContent('1');
    });

    test('fetches initial data on mount', async () => {
        // Arrange
        await act(async () => {
            render(
                <FoodProvider>
                    <div>Test Child</div>
                </FoodProvider>
            );
        });

        // Assert
        expect(ingredientService.getAllIngredients).toHaveBeenCalledTimes(1);
        expect(dishService.getAllDishes).toHaveBeenCalledTimes(1);
    });
});
