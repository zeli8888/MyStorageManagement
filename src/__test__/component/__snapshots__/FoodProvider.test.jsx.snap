// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`FoodProvider > matches snapshot 1`] = `
<div>
  <div>
    Test Child
  </div>
</div>
`;
